mathesar_ui/src/component-library/select/__meta__/Select.test.ts

```diff
@@ -180,3 +180,58 @@ test('hide select dropdown by pressing ESC', async () => {
   const selectOption2 = queryByRole('listbox');
   expect(selectOption2).not.toBeInTheDocument();
 });
+
+test('list scrolls automatically to display the currently selected option', async () => {
+  const options = [];
+  for (let i = 1; i < 10; i += 1) {
+    options.push({ id: i, label: `Option ${i}` });
+  }
+
+  const { getByRole, getAllByRole } = render(Select, {
+    props: {
+      options,
+      triggerAppearance: 'default',
+    },
+  });
+
+  // options are only rendered when the button is clicked
+  const selectBtn = getByRole('button');
+  await fireEvent.click(selectBtn);
+
+  // dropdown container obtained
+  const selectUl = getByRole('listbox');
+  const dropdownContainer: HTMLElement | null = selectUl?.parentElement;
+
+  // list of options obtained
+  const optionElementList = getAllByRole('option');
+
+  // scrolling through the list to reach second last element
+  Array(7).forEach(() => async () => {
+    await fireEvent.keyDown(selectBtn, {
+      key: 'ArrowDown',
+      code: 'ArrowDown',
+    });
+    await fireEvent.keyUp(selectBtn, { key: 'ArrowDown', code: 'ArrowDown' });
+  });
+
+  // second last element obtained
+  const optionElement = optionElementList[7];
+
+  // select the option.
+  await fireEvent.keyDown(selectBtn, { key: 'Enter', code: 'Enter' });
+  await fireEvent.keyUp(selectBtn, { key: 'Enter', code: 'Enter' });
+
+  // Reopen the dropdown
+  await fireEvent.click(selectBtn);
+
+  expect(optionElement.offsetTop).toBeGreaterThanOrEqual(
+    dropdownContainer ? dropdownContainer.scrollTop : 0,
```

Comment:
Obtaining the `dropdownContainer` element should be done before this expect statement. We also need to ensure that `dropdownContainer` is present in the dom.

This test would still pass if `dropdownContainer` is not even rendered, which should not be the case.

Reply:
@pavish 

I've done the changes:

1. dropdownContainer now gets initialized right before the assertions
2. added a check to see if dropdownContainer is in the DOM

Reply:
@pavish 

I've done the changes:

1. dropdownContainer now gets initialized right before the assertions
2. added a check to see if dropdownContainer is in the DOM